otter_welcome_buddy/cogs/new_roles.py

```diff
@@ -0,0 +1,223 @@
+import discord
+from discord import TextChannel
+from discord.ext import commands
+from discord.ext.commands import Bot
+from discord.ext.commands import Context
+
+from otter_welcome_buddy.common.constants import OTTER_ADMIN
+from otter_welcome_buddy.common.constants import OTTER_MODERATOR
+
+
+class NewRoles(commands.Cog):
+    """
+    Newroles commands who assigns roles for the users that reacts a messages
+    Commands:
+        roles add_role:     Create new message and assign roles to an emoji
+        roles edit_message: Edit a message and optionally assign a role to an emoji
+        roles edit_role:    Edit rol or emoji
+        roles new_role:     Create a message with a role assigned to an emoji
+        roles remove_role:  Remove a role from a message
+    """
+
+    _DEFAULT_MESSAGE: str = (
+        "Hola {role}, si quieres un nuevo rol para {role_name}, "
+        "reacciona a este mensaje con {emoji}."
+    )
+
+    def __init__(self, bot: Bot):
+        self.bot: Bot = bot
+        self.reaction_role_config: dict[tuple[int, str], int] = {}
+
+    @commands.group(name="roles")
+    @commands.has_any_role(OTTER_ADMIN, OTTER_MODERATOR)
+    async def roles(self, ctx: Context) -> None:
+        """Principal command"""
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+            return
+
+    @roles.command(name="add_role")  # type: ignore
+    @commands.has_any_role(OTTER_ADMIN, OTTER_MODERATOR)
+    async def add_role(
+        self,
+        ctx: Context,
+        message_id: int | None = None,
+        role_name: str | None = None,
+        emoji: str | None = None,
+    ) -> None:
+        """Asign a role to a user when they react to a message with a specific emoji"""
+        if ctx.guild is None:
+            return
+        role = discord.utils.get(ctx.guild.roles, name=role_name)
+        if role and emoji and message_id:
+            self.reaction_role_config[(message_id, emoji)] = role.id
+            await ctx.message.add_reaction("✅")
+        else:
+            await ctx.message.add_reaction("❌")
+            await ctx.send("Rol no encontrado")
+
+    @roles.command(name="edit_message")  # type: ignore
+    @commands.has_any_role(OTTER_ADMIN, OTTER_MODERATOR)
+    async def edit_message(
+        self,
+        ctx: Context,
+        message_id: int,
+        new_message: str,
+        role_name: str | None = None,
+        emoji: str | None = None,
+    ) -> None:
+        """Edit a message and optionally assign a role to an emoji"""
+        message = await ctx.fetch_message(message_id)
+        await message.edit(content=new_message)
+        if emoji and ctx.guild:
+            role = discord.utils.get(ctx.guild.roles, name=role_name)
+            if role:
+                self.reaction_role_config[(message_id, emoji)] = role.id
+                await ctx.message.add_reaction("✅")

```

Comment:
I can see that you are repeating this code multiple times, is it possible to have some common utility?